gossip/src/duplicate_shred.rs

```diff
@@ -127,6 +132,24 @@ where
         return Ok(());
     }
 
+    // Chained merkle root conflict check
+    let first_shred = std::cmp::min_by(shred1, shred2, |s1, s2| {
+        s1.fec_set_index().cmp(&s2.fec_set_index())
+    });
+    let second_shred = std::cmp::max_by(shred1, shred2, |s1, s2| {
+        s1.fec_set_index().cmp(&s2.fec_set_index())
+    });
```

Comment:
wouldn't this set `first_shred = second_shred = shred1` if both shreds have the same `fec_set_index`?